skyportal/models.py

```diff
@@ -1471,6 +1473,16 @@ class Allocation(Base):
         doc="The Instrument the allocation is associated with.",
     )
 
+    altdata = sa.Column(
+        EncryptedType(JSONType, cfg['app.secret_key'], AesEngine, 'pkcs5')
+    )
+
+    def load_altdata(self):
+        if self.altdata is None:
+            return {}
+        else:
+            return json.loads(self.altdata)
```

Comment:
why is this needed ? why can't this entire function be replaced by  `return self.altdata or {}`

Reply:
json.loads() makes sure it is a dictionary, as of now it is a string.

Reply:
OK, in that case let's handle the encryption transparently for the user by implementing as a property.

```
    _altdata = sa.Column(
        EncryptedType(JSONType, cfg['app.secret_key'], AesEngine, 'pkcs5')
    )

    @property
    def altdata(self):
        if self.altdata is None:
            return {}
        else:
            return json.loads(self.altdata)
        
    @altdata.setter
    def altdata(self, value):
        self._altdata = value
```